### ISSUE TITLE
V3 queryClient.setQueryDefaults seems to be matching incorrectly

### ISSUE THREAD

**Issue by tikotzky** [tannerlinsley/react-query#1214]
**Describe the bug**
When looking up defaults set using `queryClient.setQueryDefaults` it seems to be doing a partial match in the wrong direction.

For example, setting a more specific default will match on a generic query
```js
queryClient.setQueryDefaults(['todos', 5], queryFn);
useQuery(['todos']); // works even though the default is only set on `['todos', 5]`
```

but setting a generic default will not match on a more specific query
```js
queryClient.setQueryDefaults(['todos'], queryFn);
useQuery(['todos', 5]); // errors with cannot find queryFn
```

**To Reproduce**
You can view a sandbox with a reproduction here:
https://codesandbox.io/s/happy-snowflake-occ3q?file=/src/index.js:393-421

**Expected behavior**
a queryKey of `['todo2']` should **not** use the `queryFn` set on `queryClient.setQueryDefaults(['todos2', 5], queryFn)`
a queryKey of `['todo1', 5]` **should** use the `queryFn` set on `queryClient.setQueryDefaults(['todos1'], queryFn)`

**Comment by boschni** [comment]
Thanks for reporting! Should be fixed in `3.2.0-beta.13`.

**Comment by tikotzky** [comment]
Thanks!
Looks like its working as expected on `3.2.0-beta.13`

`3.2.0-beta.13` sandbox: https://codesandbox.io/s/crazy-antonelli-buu7k?file=/src/index.js:438-445
